R/mod_query_data.R

```diff
@@ -305,8 +305,19 @@ mod_query_data_server <- function(id, tadat) {
       )
 
       # user uploaded data
-      raw <-
-        suppressWarnings(readxl::read_excel(input$file$datapath, sheet = 1))
+      raw <- readxl::read_excel(input$file$datapath, sheet = 1, col_types = "text")
+      
+      # run autoclean
+      raw <- EPATADA::TADA_AutoClean(raw)
+      
+      #####
+      # check that all TADA template columns are included (returns TRUE or FALSE)
+      # if FALSE, returns an error with names of specific columns that are missing but required
+      # this section needs to be updated to handle the error within the shiny app (instead of crashing) & display a message to users
+      EPATADA::TADA_CheckRequiredFields(raw)
```

Comment:
@triphook I am working on related updates to the TADA package, but once those are merged and this PR is merged, then this will be a good example to look at for implementing error handling in TADAShiny. 

Here, when TADA_CheckRequiredFields runs on a dataset that is uploaded by a user (option C), but that dataset is missing required cols to run the app, then TADA_CheckRequiredFields will intentionally return an error which includes the missing cols. Instead of crashing the shiny app, this message should display for users and they should be able to edit their file and re-upload.